hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java

```diff
@@ -659,7 +660,12 @@ public IScheduledExecutorService getScheduledExecutorService(String name) {
             }
 
             for (DistributedObjectInfo distributedObjectInfo : localDistributedObjects) {
-                proxyManager.removeProxy(distributedObjectInfo.getServiceName(), distributedObjectInfo.getName());
+                String serviceName = distributedObjectInfo.getServiceName();
+                String objectName = distributedObjectInfo.getName();
+                ClientProxy proxy = proxyManager.getProxy(serviceName, objectName);
```

Comment:
Should the local destroy of the proxy not be done as part of the proxyManager.remove?